File: node/tokens/request.js
const { Ocean, Logger } = require('@oceanprotocol/squid')
const config = require('../config')
const input = require('../input');

(async () => {
    const ocean = await Ocean.getInstance(config)

    const accounts = await ocean.getAccounts()
    Logger.log(accounts)
    const account = accounts.filter(account => account.id === config.address)[0]
    Logger.log(account, input)
    await account.requestToken(input)
    // BUG: doesn't update balance
    const balance = await account.getOceanBalance()
    Logger.log(balance)
    Logger.log(`__result__${account.id} received ${input} tokens`)
})()

File: python/tokens/request.py
from squid_py.ocean.ocean import Ocean
from ..config import CONFIG_FILE
from ..input import INPUT

import logging
logging.getLogger().setLevel(logging.WARNING)

ocean = Ocean(CONFIG_FILE)
account_address = ocean.config.get('keeper-contracts', 'account.address')
account_password = ocean.config.get('keeper-contracts', 'account.password')
ocean.set_main_account(account_address, account_password)

rcpt = ocean.main_account.request_tokens(INPUT)
ocean._web3.eth.waitForTransactionReceipt(rcpt)

print("__result__{} received {} tokens".format(ocean.main_account.address, INPUT))
